geode-core/src/main/java/org/apache/geode/internal/net/BufferPool.java

```diff
@@ -190,8 +278,15 @@ void releaseBuffer(BufferPool.BufferType type, ByteBuffer buffer) {
    */
   private void releaseBuffer(ByteBuffer bb, boolean send) {
     if (useDirectBuffers) {
+
       BBSoftReference bbRef = new BBSoftReference(bb, send);
-      bufferQueue.offer(bbRef);
+      if (bb.capacity() <= Connection.SMALL_BUFFER_SIZE) {
```

Comment:
Every buffer in bufferSmallQueue should have a capacity of SMALL_BUFFER_SIZE. To make this clear you should use "==" here instead of "<="